### DISCUSSION TITLE
Can't set up a working OL. Is there a "easy to copy and paste" version available?

### DISCUSSION THREAD

**Discussion by jojo-gfx** [openlayers/openlayers#14976] [Q&A]
Was trying any methods and per npm is running against a wall everytime because there seems to be bugs in the code?
Was trying it on Ubuntu 22.04 (heavy errors) and also on Debian 11 and there seems a syntax error in the sources so it also runs aganinst a wall.
I was trying to copy the package per hand but there is the linking between the scripts not ok so also only it is running with errors.

Was copy the whole package with the examples and that is running, but don't like to use to run it as project package and was not able to set the paths correct in other directories to "ol/...".

Is there an easy way to copy a small standard package to my web directory without Git, npm and without path problems?

Git is generally a good Idea, but to push everybody into this line makes many people many problems and so many smaller projects are landing in the waste because this heavy barrier to a running the code.
Was using it years before easily but now after two days of heavy work my energy is down to make it running. :(

The users would like to concetrate on their projects and not waste their energy in problems with no running basecode.

I don't mean it bad, but we mostly are no linux nerds and no JS-nerds to debug sourcecodes. So I've seen many have problems and writing in several forums especcially about the big problem of the include-errors because "ol/" instead of "/ol/..". 
Also a tip to solving it could be helpful, denn the only solution seems to be a npm-make, but if it not runs, we all have a big big problem.

I would think a easy solution would be a copy of a project from anyone who npm is made a working standard project.
If this is ready to run, could be copied, zippen and download to own servers and ready. :) But don't know anyone who can do it for me.
Such ZIP-file on mainpage ol OL would be extreme helpful for no GIT/npm nerd users. :)

Thank you!

**Comment by mprins** [comment]
I guess you want to look into using the "hosted build", see: https://openlayers.org/download/ or you can copy the dist folder fro m the release to your webserver and use that

**Comment by jojo-gfx** [comment]
The hosted build is a risk for a running system. If there are changes on definitions or bugs so instantly my system will have a problem. So I'm not a fan of that solution. ;)

Yes, was download the Packages but there I (and others) have the problem that only examples are running, but own things have the "ol/..." problem and I don't know what I can do.

npm here has generated a site on my server, before it breaks. There is a structure generated, but also there is the script running into the "ol/" error ... seemst the script isn't ready because the error from npm.

So, copy thte whole package was also my original intension but the "ol" error prevents scripts from running, so I could modify one of the examples and running it in this folder, but this is no really solution.

Where can I define that scripts accept an include with wrong path description? The running examples have anywhere an descriptor for that definition of paths but don't find it and have also nothing found in the web so far.
Or have I overseen anything in the ZIP-package and there is a project folder who is ready to use?

**Comment by jojo-gfx** [comment]
Have now cut out what is in the "en" folder and put it on my server.
Also rename of examples folder is working.
So I try to use it so...

So far, to analyze the examples is bad, because there is all in one line.
Try to make own files is all time going against the wall because "ol" Problem.

Seems everytime is need a sort of make in the project and without them nothing is running. 
Makes no fun anymore, if software is so nailed with automatisms.

Again, I landed again in a really bad dead end.

For me it seems that users where is running npm not correct have no chance to make own OL projects?
That only because a horrible definition method in scripts overall. Why is here not used a global variable instead of wrong link definitions in input statements? So it would no problem to define it in automatisms or also easy per hand.

I don't understand the system and where I can correct it?

**Comment by ahocevar** [comment]
@jojo-gfx For proper use of the full build, go to the [release page](https://openlayers.org/download/) and download the release package (currently [v7.4.0-package.zip](https://github.com/openlayers/openlayers/releases/download/v7.4.0/v7.4.0-package.zip)). The script you'll want to reference on your website is the `./dist/ol.js`, and you also need `./ol.css`.

For the use of the npm package, follow the instructions in the [Quickstart tutorial](https://openlayers.org/doc/quickstart.html). You do not need to clone the OpenLayers repository, and you do not need to run any npm scripts from it.

**Reply by jojo-gfx** [reply]
Yes, but that is running into a syntax error: **npm start**
That's the problem and this on two systems ... Ubuntu has shown more then one errors.
It runs a while but then error and end, so the app isn't ready created.

That is, what i mean what is problematic on such structures. Anyone does anywhere a tic change the code in a version and all goes wrong after with up to there stable projects.
So I'm a fan of holding things easy for the next following level of development what is here e.g. the cartographer who has another knowledge then the developers of OL. Think the most (in tiny projects) was never using npm systems and also I have no idea, how dangerous it is to run a app with such things in the background.
Self changing code does nobody really want to host so unwanted updates are not possible. So for stable server structures I find the npm a bit uaaaaah! :D ... also because I don't know what it does without my knowledge.

This only as a little feedback from a ordinary cartographer who is no nerd in Linix, Git, npm and deeper JS. :)
Think more then 50% who like to use OL are on this level, because we have other priorities and only need OL as **easy** base.

At time it remembers me onto Typo3, what was a nice system but at time also many professionals can't mostly make good projects because the developers are in another thinking-world then they and make this and that and all are confused because nothing works anymore what everytime was working. Bad there, the base code was and is really horrible if you look behind the code. Has leading a big Typo3 System and this bad base was made the whole system slooooooow. :( ... and the work ot really expensive FE/BE developers was often runnig against a wall because too much underground API changes.
But is another thing and only a example for the often seen cliff between dev and users. 

Is only my view and not representative, but if my view could help you to overthink anything, so it would be nice! :)
**My wish: The option to download -> unzip to server -> edit index.html/main.js -> ready(!) would be a perfect thing! ;)**
As far I know, that was also the goal years before, or?

**Comment by jojo-gfx** [comment]
Aaah, this looks good and seems to be what I want! :)
I try it ... thanx!

**Comment by jojo-gfx** [comment]
Seems the ol.js has a bug!?
The first line ends with: var qm={
So can't use it so far ... and also repair isn't possible, because all is in one line.

Or how do I use it? Is there a description anywhere for all who npm dont't work?
npm was display a syntax error in source so far I know...

**Comment by ahocevar** [comment]
@jojo-gfx That's minified. Everything is in a global `ol` object. In the API docs, you have the imports documented, e.g. `import Vector from "ol/layer/Vector"` translates to `ol.layer.Vector`.

And yes, the developer experience with npm is much, much better for sure. It's definitely worth getting famiiar with it. Not only for OpenLayers, but for modern JavaScript development in general.

**Comment by jojo-gfx** [comment]
Yes, no bad thing, but when it doesn't run correct on users machine so it is no big deal. ;)
On two several machines it goes wrong and no idea, how I could fix it? With GIT in general have never had problems, but this will not run here.

I think I'll give up, because I'm to stupid to manage this amount an errors and code manipulations. I have had an Idea and was on the way to start a OL, so easy like it was for years, but my whole energy is away now and think about another solution.

It would be nice if there would be a running copy and paste package for small projects to download and run out of the box.
But at time it is the overkill and seems only to handle for bigger institutions with many ressources.

Or is there an old version, who is running in the normal way? So I will try to use this but don't know when was the point of switching to this structure?

Really not only my whish: A simple copy and paste package. ;)
I have seen many frustrated users in forums who have the same problem with no solution at the end of the threads. ;(

**Comment by ahocevar** [comment]
Since you haven't shown any of the resources you've been using to get started, none of the errors you got, and none of the forums you've been looking at, it is not easy for any of the volunteers here to help.

If what you want is a zip containing a website with a working OpenLayers map, where you can edit `index.html` and `main.js`, then this is for you:

[ol-example.zip](https://github.com/openlayers/openlayers/files/12267355/ol-example.zip)

**Reply by jojo-gfx** [reply]
Thanks, in princip yes and I'll try again to find a way to use it.
I have any GeoJson files to bind in and will try to include the data in hope, there are no inner "ol/..." links to other sources. ;) 

------
Was looking per google about the special error with the failed links and have found many sites. Can't say more but was my impression I'm not alone. But could be I misunderstand anything here... sorry! Without the errors in npm start all would be ok, but wasn't.

**Reply by ahocevar** [reply]
Note that when you get errors upon `npm start`, these don't primarily come from your dependencies, but from your application code.

**Reply by jojo-gfx** [reply]
Was try it again and also with the first command line wa any suspect and seems the problem. But I have no ideas here:

#npm create ol-app map1

npm WARN EBADENGINE Unsupported engine {
npm WARN EBADENGINE   package: 'rollup@3.10.0',
npm WARN EBADENGINE   required: { node: '>=14.18.0', npm: '>=8.0.0' },
npm WARN EBADENGINE   current: { node: 'v12.22.12', npm: '7.5.2' }
npm WARN EBADENGINE }
npm WARN EBADENGINE Unsupported engine {
npm WARN EBADENGINE   package: 'vite@4.0.4',
npm WARN EBADENGINE   required: { node: '^14.18.0 || >=16.0.0' },
npm WARN EBADENGINE   current: { node: 'v12.22.12', npm: '7.5.2' }
npm WARN EBADENGINE }

added 66 packages, and audited 67 packages in 3s

6 packages are looking for funding
  run `npm fund` for details

1 high severity vulnerability

To address all issues, run:
  npm audit fix

Run `npm audit` for details.

------
..... npm audit was saying "found 0 ...." so seems to be ok?

**Reply by jojo-gfx** [reply]
Seems the node is in wrong version. I try to find a solution ...

So no wonder that npm start no is running perfect after that.

**Comment by ahocevar** [comment]
On any Linux system, my suggestion is to use [nvm](https://github.com/nvm-sh/nvm) to install node, rather than the packages of the Linux distribution. Regarding node version, it's always a good idea to use the latest LTS version. At the time of writing, this is version 18. With nvm, you simply type `nvm install 18` to get that version, and `nvm use 18` to use it.

**Reply by jojo-gfx** [reply]
Ah, yes, have found it already and install. 
Now npm works ... uaaaah! :D 
After generating all and also seems successful build, only errors on test in browser, but I'll read first, what could be wrong here.

Thank you for your understanding! ;) Is all new for me to use npm and so on ... 

Greetings and thanx!

**Reply by jojo-gfx** [reply]
Aaaah! Successs... now I understand what I have to do... :D
